crates/common/src/raindex_client/local_db/sync.rs

```diff
@@ -146,10 +146,41 @@ impl RaindexClient {
             .map_err(|e| LocalDbError::FetchEventsFailed(Box::new(e)))?;
 
         send_status_message(&status_callback, "Decoding fetched events...".to_string())?;
-        let decoded_events = local_db
+        let mut decoded_events = local_db
             .decode_events(&events)
             .map_err(|e| LocalDbError::DecodeEventsFailed(Box::new(e)))?;
 
+        let mut store_addresses: HashSet<String> = collect_store_addresses(&decoded_events)
+            .into_iter()
+            .collect();
+
+        let existing_stores: Vec<StoreAddressRow> =
+            LocalDbQuery::fetch_store_addresses(&db_callback).await?;
+        for row in existing_stores {
+            if !row.store_address.is_empty() {
+                store_addresses.insert(row.store_address.to_ascii_lowercase());
+            }
+        }
```

Comment:
nitpick: consider the following (once again, would be even simpler with `Address` as neither the filtering nor the mapping step would be needed)

```suggestion
        let existing_stores: Vec<StoreAddressRow> =
            LocalDbQuery::fetch_store_addresses(&db_callback).await?;

        store_addresses.extend(
            existing_stores
                .into_iter()
                .filter(|row| !row.store_address.is_empty())
                .map(|row| row.store_address.to_ascii_lowercase())
        );
```